src/Exceptions/InvalidIndexException.php

```diff
@@ -1,19 +0,0 @@
-<?php
-
-namespace Spatie\Enum\Exceptions;
-
-use InvalidArgumentException;
-
-class InvalidIndexException extends InvalidArgumentException
```

Comment:
I think that we still need this exception but extend it to also allow `string` for `value`.